pkg/controllers/clusteroperator_controller.go

```diff
@@ -251,7 +269,77 @@ func (r *CloudOperatorReconciler) provisioningAllowed(ctx context.Context, infra
 		return false, nil
 	}
 
-	// TODO: Add part confirming migration is finished for KCM before allowing provisioning
+	ownedByKCM, err := r.isCloudControllersOwnedByKCM(ctx)
+	if err != nil {
+		return false, err
+	}
+	if ownedByKCM {
+		// KCM resource found and it owns Cloud provider
+		klog.Infof("Kube Controller Manager still owns Cloud Controllers. Skipping...")
+
+		if err := r.setStatusAvailable(ctx); err != nil {
+			klog.Errorf("Unable to sync cluster operator status: %s", err)
+			return false, err
+		}
+
+		return false, nil
+	}
 
 	return true, nil
 }
+
+func (r *CloudOperatorReconciler) isCloudControllersOwnedByKCM(ctx context.Context) (bool, error) {
+	kcm := &operatorv1.KubeControllerManager{}
+	err := r.Get(ctx, client.ObjectKey{Name: kcmResourceName}, kcm)
+	if err != nil {
+		klog.Errorf("Unable to retrive Kube Config Manager object: %v", err)
+
+		if err := r.setStatusDegraded(ctx, err); err != nil {
+			klog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
+			return false, fmt.Errorf("error syncing ClusterOperatorStatus: %v", err)
+		}
+		return false, err
+	}
+
+	if len(kcm.Status.Conditions) == 0 {
+		// If KCM Conditions object doesn't exist, we assume that we are in bootstrapping process and
+		// the controllers are not owned by KCM.
+		klog.Info("Kube Config Manager status not found. It's considered that KCM does not own the controllers.")
```

Comment:
I think this might be a better message that explains why we are going ahead and deploying
```suggestion
		klog.Info("KubeControllerManager status not found, cluster is bootstrapping with external cloud providers")
```